specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/registries.json

```diff
@@ -0,0 +1,873 @@
+              },
+              "Retry-After": {
+                "description": "Duration the client should wait between requests, in seconds.",
+                "type": "integer",
+                "format": "int32",
+                "maximum": 600,
+                "minimum": 10
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/RegistryTrackedResource"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Update Registry with system created accounts.": {
+            "$ref": "./examples/Registries/update-SystemCreated.json"
+          },
+          "Update Registry with user created accounts.": {
+            "$ref": "./examples/Registries/update-UserCreated.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "RegistryManagement"
+        ],
+        "summary": "Create or update registry",
+        "operationId": "Registries_CreateOrUpdate",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "in": "path",
+            "name": "registryName",
+            "description": "Name of registry. This is case-insensitive",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9\\-_]{0,254}$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Details required to create the registry.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RegistryTrackedResource"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/RegistryTrackedResource"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/RegistryTrackedResource"
+            }
+          },
+          "202": {
+            "description": "Created",
```

Comment:
PUT doesnt need 202. can be removed

Reply:
this was added in based on [TASK 2036303: Document registry swagger changes that are not in the published swagger yet](https://msdata.visualstudio.com/Vienna/_workitems/edit/2036303) -> "Changed registry delete, create, and update operations to accept a 202 response code"

Reply:
Not a blocking comment.

Reply:
It really is not recommended to use the 202 model for a PUT , this is explained in detail here : [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)




IF you choose to disregard and decide to return 202 , it should not return a body : 
[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#202-accepted-and-location-headers](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#202-accepted-and-location-headers)


IN addition to azure guidelines , this model also violates general microsoft guidelines as well : 
[https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#1321-put](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#1321-put)

Reply:
removed body.